src/bus/policy.h

```diff
@@ -162,16 +166,21 @@ int policy_snapshot_dup(PolicySnapshot *snapshot, PolicySnapshot **newp);
 int policy_snapshot_check_connect(PolicySnapshot *snapshot);
 int policy_snapshot_check_own(PolicySnapshot *snapshot, const char *name);
 int policy_snapshot_check_send(PolicySnapshot *snapshot,
-                               const char *subject_context,
+                               const char *sender_context,
```

Comment:
The sender context should be taken from `snapshot` (see `snapshot->seclabel`). Otherwise, you will query the context too late. Note that the point of the policy-snapshot is to snapshot all sender information at a particular time, and then *later* check whether it matches. This is used for activation messages where sender and receiver information is *not* available simultaneously. If this does not work for apparmor, this needs to be somehow documented.